From c96501b3e28bc52b1912f1b53b42cceefe2e5371 Mon Sep 17 00:00:00 2001
From: Hugo Villeneuve <hvilleneuve@dimonoff.com>
Date: Thu, 11 Apr 2024 15:57:11 -0400
Subject: [PATCH 2/3] rzg2: add generic distro boot support

Signed-off-by: Hugo Villeneuve <hvilleneuve@dimonoff.com>
---
 include/configs/smarc-rzg2-common.h | 36 ++++++++++++++++++++++-------
 1 file changed, 28 insertions(+), 8 deletions(-)

diff --git a/include/configs/smarc-rzg2-common.h b/include/configs/smarc-rzg2-common.h
index b5da03c39a..e24ed5485e 100644
--- a/include/configs/smarc-rzg2-common.h
+++ b/include/configs/smarc-rzg2-common.h
@@ -55,19 +55,39 @@
 /* The HF/QSPI layout permits up to 1 MiB large bootloader blob */
 #define CONFIG_BOARD_SIZE_LIMIT		1048576
 
+/* Distro boot settings */
+
+#define BOOT_TARGET_DEVICES(func) \
+        func(MMC, mmc, 0) \
+        func(USB, usb, 0) \
+        func(DHCP, dhcp, na)
+
+#include <config_distro_bootcmd.h>
+
+#define KERNEL_ADDR_R       __stringify(0x48000000)
+#define KERNEL_COMP_ADDR_R  __stringify(0x4a000000)
+#define FDT_ADDR_R          __stringify(0x4c000000)
+#define SCRIPT_ADDR_R       __stringify(0x4c100000)
+#define PXEFILE_ADDR_R      __stringify(0x4c200000)
+#define RAMDISK_ADDR_R      __stringify(0x4c800000)
+#define KERNEL_COMP_SIZE    __stringify(0xb00000)
+
 /* ENV setting */
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"usb_pgood_delay=2000\0" \
 	"bootm_size=0x10000000 \0" \
-	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
-	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
-	"bootimage=unzip 0x4A080000 0x48080000; booti 0x48080000 - 0x48000000 \0" \
-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/" KERNEL_DT_BASENAME ".dtb;run prodemmcbootargs \0" \
-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/" KERNEL_DT_BASENAME ".dtb;run prodsdbootargs \0" \
-	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
-
-#define CONFIG_BOOTCOMMAND	"env default -a;run bootcmd_check;run bootimage"
+	"kernel_addr_r=" KERNEL_ADDR_R "\0"   \
+	"fdt_addr_r=" FDT_ADDR_R "\0" \
+	"ramdisk_addr_r=" RAMDISK_ADDR_R "\0" \
+	"scriptaddr=" SCRIPT_ADDR_R "\0" \
+	"pxefile_addr_r=" PXEFILE_ADDR_R "\0" \
+	"fdt_high=0xffffffffffffffff\0"     \
+	"initrd_high=0xffffffffffffffff\0" \
+	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
+	"kernel_comp_addr_r=" KERNEL_COMP_ADDR_R "\0" \
+	"kernel_comp_size=" KERNEL_COMP_SIZE "\0" \
+BOOTENV
 
 /* For board */
 /* Ethernet RAVB */
-- 
2.39.2

